### PR TITLE
Replace update_java_tools.sh with several bazel targets.

### PR DESCRIPTION
**Pull request by iirina**
Remove the cumbersome script `third_party/java/java_tools/update_java_tools.sh` and replace it with bazel targets that build the java tools deploy jars. `third_party:java_tools_zip` zips the deploy jars built at the current commit hash.

It's unnecessary to keep around the deploy jars on Github, since they are already stored inside the Java tools zip on GCS.

This PR makes it much easier to release the Java tools, by running the newly set up [Buildkit pipeline](https://buildkite.com/bazel-trusted/java-tools-binaries-java) that builds `third_party:java_tools_zip` and uploads it automatically to GCS (for all 3 platforms).

### CODE REVIEW

**Comment by iirina** [comment]
@lberki ping :)

File: src/java_tools/junitrunner/java/com/google/testing/coverage/BUILD [hunk 1]
```diff
@@ -63,4 +63,8 @@ genrule(
         "//third_party/jarjar:jarjar_command_deploy.jar",
         "@bazel_tools//tools/jdk:current_host_java_runtime",
     ],
+    visibility = [
+        "//src:__subpackages__",
```
Comment (lberki) [inline src/java_tools/junitrunner/java/com/google/testing/coverage/BUILD]: nit: `//src:__subpackages__` is quite a few packages. Why is this necessary?

File: third_party/BUILD [hunk 2]
```diff
@@ -623,6 +623,22 @@ filegroup(
...
+        "//src/java_tools/buildjar:srcs",
+        "//src/java_tools/junitrunner:srcs",
+        "//src/java_tools/singlejar:srcs",
+        "//third_party/jarjar:srcs",
+        "//third_party/java/jdk/langtools:license-and-srcs",
+        "//third_party/zlib:embedded_tools",
+        "//third_party/ijar:srcs_with_deps",
+    ],
+    outs = ["java_tools_dist.zip"],
+    cmd = "zip -Xr $@ $$(echo $(SRCS) | sort)",
```
Comment (lberki) [inline third_party/BUILD]: nit: +`-9`?

Comment (lberki) [inline third_party/BUILD]: Do we want to null out the timestamps?

File: third_party/BUILD [hunk 3]
```diff
@@ -623,6 +623,22 @@ filegroup(
...
+    output_to_bindir = 1,
```
Comment (lberki) [inline third_party/BUILD]: Why is this necessary?

File: third_party/ijar/BUILD [hunk 4]
```diff
@@ -94,6 +94,12 @@ filegroup(
     visibility = ["//third_party:__pkg__"],
 )
 
+filegroup(
+    name = "srcs_with_deps",
```
Comment (lberki) [inline third_party/ijar/BUILD]: nit: `transitive_sources`?

File: third_party/java/java_tools/BUILD [hunk 5]
```diff
@@ -30,3 +19,122 @@ genrule(
...
+        "turbine_deploy.jar",
+        "turbine_direct_binary_deploy.jar",
+    ] + select({
+        "//src/conditions:arm": ["bazel-singlejar_deploy.jar"],
+        "//conditions:default": [],
+    }),
+    visibility = ["//visibility:private"],
+)
+
+genrule(
```
Comment (lberki) [inline third_party/java/java_tools/BUILD]: What are these genrules for?

File: src/java_tools/junitrunner/java/com/google/testing/coverage/BUILD (hunk 1 above)
Comment (iirina) [reply src/java_tools/junitrunner/java/com/google/testing/coverage/BUILD]: `//src:__subpackages__` is the default visibility for this package. I want to preserve the default visibility and add another one, otherwise the visibility would be overwritten.

File: third_party/BUILD (hunk 2 above)
Comment (iirina) [reply third_party/BUILD]: I tried archiving using `-9` and it doesn't make a difference...

Comment (iirina) [reply third_party/BUILD]: Yup, striving for determinism...

File: third_party/BUILD (hunk 3 above)
Comment (iirina) [reply third_party/BUILD]: Hm just for consistency with the other java_tools_zip rule... It's not a hard requirement.

File: third_party/ijar/BUILD (hunk 4 above)
Comment (iirina) [reply third_party/ijar/BUILD]: Thanks :)

File: third_party/java/java_tools/BUILD (hunk 5 above)
Comment (iirina) [reply third_party/java/java_tools/BUILD]: You're right, I don't need them. I removed them

File: third_party/BUILD (hunk 2 above)
Comment (lberki) [reply third_party/BUILD]: ...but are they nulled out here?

File: third_party/ijar/BUILD [hunk 6]
```diff
@@ -94,6 +94,12 @@ filegroup(
     visibility = ["//third_party:__pkg__"],
 )
 
+filegroup(
```
Comment (lberki) [inline third_party/ijar/BUILD]: nit: do we need the `srcs` filegroup then?

Comment (iirina) [reply third_party/ijar/BUILD]: Yes, it's used to build the dist archive. Why shouldn't we need it?
